### PR TITLE
Explore: Fix timezone not changing for graph and table

### LINKED ISSUE #39279: Explore: Graph and Table don't change times according to timezone
**What happened**:

When updating timezone in time picker, graph panel and table panel don't update their times according to the timezone. 

**What you expected to happen**:
The time should be changed based on timezone as we do it in dashboard and logs

**How to reproduce it (as minimally and precisely as possible)**:

Prometheus - graph and table don't change times based on timezone:

https://user-images.githubusercontent.com/30407135/133575732-596cf4a7-9888-4927-bfdb-662fecaabdf0.mov

Prometheus - graph doesn't change times based on timezone:

https://user-images.githubusercontent.com/30407135/133575908-87328428-bc58-47fc-9670-cb3f82f02c5f.mov

**Anything else we need to know?**:

**Environment**:
- Grafana version: master (8.2-pre)

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/39279

### CODE REVIEW

**Review by Elfo404 — COMMENTED** [review]
Didn't manually test this but the tests seem to cover everything (stellar work on these ⭐).

wdyt about using `InternalTimeZones.utc` instead of hardcoding `utc` in the tests? (also might be more reliable given that internally `InternalTimeZones.utc` is `UTC`)

File: public/app/features/explore/TableContainer.test.tsx [hunk 1]
```diff
@@ -1,41 +1,102 @@
...
+      fields: [],
+      length: 0,
+    } as DataFrame;
+    render(<TableContainer {...defaultProps} tableResult={emptyFrames} />);
+    expect(screen.getByText('0 series returned')).toBeInTheDocument();
+  });
+
+  it('should update time when timezone changes', () => {
+    const { rerender } = render(<TableContainer {...defaultProps} />);
+    let rowsBeforeChange = within(getTable()).getAllByRole('row');
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/features/explore/TableContainer.test.tsx:85]: Oh yes, good catch!

File: public/app/features/explore/ExploreQueryInspector.test.tsx [hunk 2]
```diff
@@ -33,6 +33,7 @@ const setup = (propOverrides = {}) => {
     width: 100,
     exploreId: ExploreId.left,
     onClose: jest.fn(),
+    timeZone: 'UTC',
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/features/explore/ExploreQueryInspector.test.tsx:36]: Yeah, that is a good idea!  I'll update.

**Review by Elfo404 — APPROVED** [review]
GR8! 🚀

**Pull request merged**